### ISSUE TITLE
kafkaTemplate in KafkaItemWriter should be protected

### ISSUE THREAD

**Issue by benas** [spring-projects/spring-batch#3802]
As of v4.3.0, the `kafkaTemplate` field in `KafkaItemWriter` is private. This makes it impossible to use the template in overridden methods like `writeKeyValue` or `write`:

```java
KafkaItemWriter<String, String> myCustomKafkaItemWriter = new KafkaItemWriter<String, String>() {
	@Override
	public void write(List<? extends String> items) throws Exception {
		// unable to use super.kafkaTemplate here
	}
};
```

The visibility of the field `kafkaTemplate` in `KafkaItemWriter` should be changed from `private` to `protected`.

**Comment by adityausathe** [comment]
I have opened PR #3807 to address this. @benas please review.

**Comment by benas** [comment]
Resolved with #3807 .
